### DISCUSSION TITLE
Javascript code not triggering ?

### DISCUSSION THREAD

**Discussion by LudwigMeert** [AutomaApp/automa#170] [Q&A] [answered]
Hello !

I don't understand why my javascript block isn't triggering :
```
window.post = function(url, data) {
  return fetch(url, {method: "POST", headers: {'Content-Type': 'application/json'}, body: JSON.stringify(data)});
}

var array=document.querySelectorAll("#search .g a");
len_array= array.length;
url_list=[];

for (let i = 0; i < 3; i++) {
  var url="";
  url=array[i].href;
  url_list.push(url);
}

console.log(url_list);

post("https://hook.integromat.com/usxlsyiux5mhmas9x58wq6214ao6gpvc", {"url_list": url_list});

automaNextBlock();
```

I've launched it from the chrome console and it works fine but somehow it doesn't execute from the javascript block (see image below).

![image](https://user-images.githubusercontent.com/95912743/145558878-3e8c957d-63ac-45e9-a60a-990c94ad2cdf.png)

Any chance you may help me understand why ?

**Comment by Kholid060** [comment, marked as answer]
Can you screenshot the console tab of the website where you execute the javascript code block?

**Reply by LudwigMeert** [reply]
It's all good, sorry for bothering you.
The workflow opens Google, searches something and then returns the first organic links... and I've just realized my code block was triggered just before the search results would load which prevented it from executing properly.

For anyone having trouble, adding a delay before firing the script did the trick. Just enough time for the new page to load before the script is fired.
